core/server/master/src/main/java/alluxio/master/file/meta/cross/cluster/CrossClusterMountSubscriber.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.meta.cross.cluster;
+
+import alluxio.client.cross.cluster.CrossClusterClient;
+import alluxio.grpc.ClusterId;
+import alluxio.proto.journal.CrossCluster.MountList;
+import alluxio.resource.LockResource;
+
+import io.grpc.stub.ClientCallStreamObserver;
+import io.grpc.stub.ClientResponseObserver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Subscribes to the cross cluster configuration service
+ * {@link alluxio.master.cross.cluster.CrossClusterState}, receiving notifications when
+ * a cross cluster enabled mount changes at an external cluster.
+ * This class runs a thread that will maintain a connection to the configuration service.
+ */
+public class CrossClusterMountSubscriber implements Closeable {
```

Comment:
This thread is needed the keep the local Alluxio cluster up to date when the CrossCluster master service goes up or down, allowing the thread to reconnect to the new master.